jerry-core/ecma/operations/ecma-reference.c

```diff
@@ -120,6 +141,10 @@ ecma_op_resolve_reference_value (ecma_object_t *lex_env_p, /**< starting lexical
         return prop_value;
       }
     }
+    else
+    {
+      JERRY_ASSERT (lex_env_type == ECMA_LEXICAL_ENVIRONMENT_SUPER_OBJECT_BOUND);
```

Comment:
Actually, this check seems to be correct only if ES2015 is enabled. Otherwise, if we reach this point, we are in trouble. (In ES5 mode, the environment type should not be super-bound, as there is no super.) This should be rather something like:

```c
#ifndef CONFIG_DISABLE_ES2015_CLASS
    JERRY_ASSERT (lex_env_type == ECMA_LEXICAL_ENVIRONMENT_SUPER_OBJECT_BOUND);
#else /* CONFIG_DISABLE_ES2015_CLASS */
    JERRY_UNREACHABLE ();
#endif /* !CONFIG_DISABLE_ES2015_CLASS */
```

Reply:
Good catch!